cabby/model/text/model.py

```diff
@@ -0,0 +1,172 @@
+# Copyright 2020 The Flax Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""LSTM classifier model for SST-2."""
+
+import functools
+from typing import Any, Callable, Dict, Text
+
+import flax
+from flax import nn
+import jax
+import jax.numpy as jnp
+from jax import lax
+
+import numpy as np
+
+# pylint: disable=arguments-differ,too-many-arguments
+
+
+@functools.partial(jax.jit, static_argnums=(0, 1, 2, 3))
+def create_model(seed: int, batch_size: int, max_len: int,
+                 model_kwargs: Dict[Text, Any]):
+  """Instantiates a new model."""
+  module = TextClassifier.partial(train=False, **model_kwargs)
+  _, initial_params = module.init_by_shape(
+      jax.random.PRNGKey(seed),
+      [((batch_size, max_len), jnp.int32),
+       ((batch_size,), jnp.int32)])
+  model = nn.Model(module, initial_params)
+  return model
+
+
+def word_dropout(inputs: jnp.ndarray, rate: float, unk_idx: int, 
+        deterministic: bool = False):
+  """Replaces a fraction (rate) of inputs with <unk>."""
+  if deterministic or rate == 0.:
+    return inputs
+
+  mask = jax.random.bernoulli(nn.make_rng(), p=rate, shape=inputs.shape)
```

Comment:
not sure - but is this costly? can we init it once as a constant and then permute for every example?